server/py/framework/db/sqldb/db.py

```diff
@@ -5143,7 +5142,13 @@ def _get_mep_latest_instance(
         if _get_query:
             return query
 
-        return query.first()  # Use `.first()` instead of `.one_or_none()` for safety
+        if query.count() > 1:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "More than one instance found for the given Model Endpoint."
+                "Please provide endpoint_id or function name and tag to get a specific instance."
+            )
+
+        return query.one_or_none()
```

Comment:
I suggest not implementing like this, as you are running 2 queries against the DB which is less efficient. Since `one_or_none()` you can use it and then compare with `None`. Or better yet use `one()` and catch the exception if it raises one, then you can raise your exception from it.

Reply:
Actually I'm wrong as you want to support 0, 1 and an exception on more than 1. So ignore my comment :)